x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/asset_criticality_service.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash/fp';
+import type { AssetCriticalityRecord } from '../../../../common/api/entity_analytics';
+import type { AssetCriticalityDataClient } from './asset_criticality_data_client';
+
+interface CriticalityIdentifier {
+  id_field: string;
+  id_value: string;
+}
+
+interface IdentifierValuesByField {
+  [idField: string]: string[];
+}
+
+export interface AssetCriticalityService {
+  getCriticalitiesByIdentifiers: (
+    identifiers: CriticalityIdentifier[]
+  ) => Promise<AssetCriticalityRecord[]>;
+}
+
+const isCriticalityIdentifierValid = (identifier: CriticalityIdentifier): boolean =>
+  !isEmpty(identifier.id_field) && !isEmpty(identifier.id_value);
+
+const groupIdentifierValuesByField = (
```

Comment:
cc: @nkhristinin 

I think the alert enrichment code does similar stuff

Reply:
Are you saying we should reuse this (or the other enrichment method)? I didn't see an analogous function in the enrichment stuff; could you please link it here?